CIScripts/Build/BuildFunctions.ps1

```diff
@@ -203,7 +203,45 @@ function Invoke-AgentBuild {
                      -MSIPath $agentMSI
 
     $Job.Step("Copying artifacts to $OutputPath", {
-        Copy-Item $agentMSI $OutputPath -Recurse -Container
+        Copy-Item $agentMSI $OutputPath -Recurse
+    })
+
+    $Job.PopStep()
+}
+
+function Invoke-NodemgrBuild {
+    Param ([Parameter(Mandatory = $true)] [string] $OutputPath,
+           [Parameter(Mandatory = $true)] [string] $LogsPath,
+           [Parameter(Mandatory = $false)] [string] $BuildMode = "debug")
+
+    $Job.PushStep("Nodemgr build")
+
+    $Job.Step("Building nodemgr", {
+        $Components = @(
+            "database:node_mgr",
+            "build/debug/sandesh/common/dist",
+            "sandesh/library/python:pysandesh",
+            "contrail-nodemgr",
+            "vrouter:node_mgr"
+        )
+        $ComponentsString = $Components -Join " "
+
+        Invoke-NativeCommand -ScriptBlock {
+            Invoke-Expression "scons $ComponentsString" | Tee-Object -FilePath $LogsPath/build_nodemgr.log
```

Comment:
Shouldn't we pass `$BuildMode` here somewhere?

You can use `scons @Components`, like here https://github.com/Juniper/contrail-windows-ci/blob/cc88d3728c344d7b43e72991402138afb7f2ed7b/Test/TestConfigurationUtils.ps1#L502